lib/pages/search/public_room/empty_search_public_room_widget.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:fluffychat/pages/search/public_room/empty_search_public_room_widget_style.dart';
+import 'package:fluffychat/pages/search/public_room/search_public_room_view_style.dart';
+import 'package:fluffychat/widgets/twake_components/twake_text_button.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+class EmptySearchPublicRoomWidget extends StatelessWidget {
+  final String genericSearchTerm;
+  final VoidCallback? onTapJoin;
+
+  const EmptySearchPublicRoomWidget({
+    super.key,
+    required this.genericSearchTerm,
+    this.onTapJoin,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: SearchPublicRoomViewStyle.paddingListItem,
+      child: Padding(
+        padding: SearchPublicRoomViewStyle.paddingInsideListItem,
+        child: Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Padding(
+              padding: SearchPublicRoomViewStyle.paddingAvatar,
+              child: ClipRRect(
+                borderRadius:
+                    EmptySearchPublicRoomWidgetStyle.avatarBorderRadius,
+                child: Container(
+                  width: EmptySearchPublicRoomWidgetStyle.avatarSize,
+                  height: EmptySearchPublicRoomWidgetStyle.avatarSize,
+                  decoration: BoxDecoration(
+                    borderRadius:
+                        EmptySearchPublicRoomWidgetStyle.avatarBorderRadius,
+                    color:
+                        EmptySearchPublicRoomWidgetStyle.avatarBackgroundColor(
+                      context,
+                    ),
+                    border:
+                        EmptySearchPublicRoomWidgetStyle.avatarBorder(context),
+                  ),
+                  child: Center(
+                    child: Text(
+                      genericSearchTerm[0],
```

Comment:
check the `genericSearchTerm` is length bigger than 0 before use

Reply:
updated